### ISSUE TITLE
Renders wrong config 

### ISSUE THREAD

**Issue by ph-t-G** [Icinga/icingaweb2-module-director#2289]
## Expected Behavior

Deploying a new cofiguration after an Upgrade led to this error:

```
critical/config: Error: Dependency "....." references a parent host/service which doesn't exist.
Location: in [stage]/zones.d/director-global/dependency_apply.conf: 8:1-8:43
[stage]/zones.d/director-global/dependency_apply.conf(6): }
[stage]/zones.d/director-global/dependency_apply.conf(7): 
[stage]/zones.d/director-global/dependency_apply.conf(8): apply Dependency "ApacheVsVhost" to Service {
                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[stage]/zones.d/director-global/dependency_apply.conf(9):     import "Dependency Template"
[stage]/zones.d/director-global/dependency_apply.conf(10): 
```

The dependency_apply.conf should look like this. 
```
apply Dependency "TomcatVsAJPPort" to Service {
    import "Dependency Template"
    assign where host.vars.tomcat_ajps && match("AJP Port *", service.name)
    parent_service_name = service.vars.parent_service_name
}
```

## Current Behavior

![image](https://user-images.githubusercontent.com/66112099/108355180-c49f6b00-71ea-11eb-9d31-30ea7c3741d8.png)

But the config looks like this.

```
apply Dependency "TomcatVsAJPPort" to Service {
    import "Dependency Template"
    assign where host.vars.tomcat_ajps && match("AJP Port *", service.name)
    parent_service_name = "$service.vars.parent_service_name$"
}


```

## Your Environment

* Director version (System - About): Current Master (e2a88861487e9cefad94776fd610b3a9703b9e03)
* Icinga Web 2 version and modules (System - About): 2.8.3
* Icinga 2 version (`icinga2 --version`): 2.12.0-506
* Operating System and version: Centos 7
* Webserver, PHP versions: Apache, PHP 7.3

**Comment by Nathaniel-Donahue** [comment]
ref/NC/702512

**Comment by lippserd** [comment]
Hi,

The error from the logs is about dependency `ApacheVsVhost`. You've pasted config for dependency `TomcatVsAJPPort`.
Is `TomcatVsAJPPort` also rendered incorrectly? Could you please post the relevant logs.

All the best,
Eric

**Comment by friesoft** [comment]
Yes the TomcatVsAJPPort is also rendered incorrectly - same problem, log looks exactly the same except the exchanged names.

**Comment by lippserd** [comment]
I'm wondering whether and how this has worked before. I've checked version 1.7.2. Director does not seem to support custom vars for `parent_service_name`. Maybe the apply rule was rendered incorrectly before so that it did not match any objects.

**Comment by stevie-sy** [comment]
There is still an open issue https://github.com/Icinga/icingaweb2-module-director/issues/1142

**Comment by lippserd** [comment]
> There is still an open issue #1142

True, but that one is about selecting an actual service where Director uses `parent_service_id`. Here, Director uses `parent_service_by_name` which is just set to a simple string and does not check whether the string pretends to be a custom variable.

**Comment by stevie-sy** [comment]
Ok, we are still waiting for the fix. So hopefully both methods will work soon :-)
